### PR TITLE
Deprecate access to single connections

### PR DESCRIPTION
**Pull request by fmoor**
The client APIs are moving toward a pool first approach. This is the first step in that direction.

### CODE REVIEW

File: docs/api/types.rst [hunk 1]
```diff
@@ -97,20 +97,20 @@ EdgeDB ``array``  maps onto the JavaScript ``Array``.
     const edgedb = require("edgedb");
 
     async function main() {
-      const conn = await edgedb.connect({
+      const pool = await edgedb.createPool({
```
Comment (elprans) [inline docs/api/types.rst:100]: I think we should continue to call this `connect()`.  Technically, this is a backwards-compatible change as all the methods behave the same. @1st1?

Comment (1st1) [reply docs/api/types.rst:100]: +1

File: src/client.ts [hunk 2]
```diff
@@ -120,6 +120,11 @@ export default function connect(
   dsn?: string | ConnectConfig | null,
   options?: ConnectConfig | null
 ): Promise<Connection> {
+  // tslint:disable-next-line: no-console
+  console.warn(
+    "The `connect()` function is deprecated and is scheduled to be " +
+      "removed. Use `createPool()` instead."
+  );
```
Comment (1st1) [inline src/client.ts:127]: Yeah, to be clear we'll actually deprecate the `createPool` function in the future, leaving only `connect()`. @elprans

Comment (1st1) [reply src/client.ts:127]: 1. should not
2. should not

this PR *should only* deprecate `Pool.acquire` and `Pool.release`. Same in edgedb-python. Everything else is a bit too complex to do right now.

File: docs/api/types.rst (hunk 1 above)
Comment (fmoor) [reply docs/api/types.rst]: https://github.com/edgedb/edgedb-js/pull/132#discussion_r686373221

File: src/client.ts (hunk 2 above)
Comment (fmoor) [reply src/client.ts]: changed. 

I also deprecated `Pool.run()` because it seemed very similar to `acquire()`.

**Review by 1st1 — APPROVED** [review]

**Pull request merged**